backend/hct_mis_api/apps/payment/models.py

```diff
@@ -1016,6 +1016,43 @@ def get_program(self) -> "Program":
         # TODO will update after add feature with 'program_cycle' and migrate all data
         return self.program_cycle.program if self.program_cycle else self.program
 
+    def _get_last_approval_process_data(self) -> dict:
+        approval_process = hasattr(self, "approval_process") and self.approval_process.first()
+        if approval_process:
+            if self.status == PaymentPlan.Status.IN_APPROVAL:
+                return {
+                    "modified_date": approval_process.sent_for_approval_date,
```

Comment:
minor thing, but this is a good place for dataclass usage instead of redefining same dict in all places 

{ "modified_date": x,"modified_by": x }


also access attribute would be not hardcoded

 return self._get_last_approval_process_data().modified_date
 
 instead of
  return self._get_last_approval_process_data()["modified_date"]